src/main/java/com/redhat/sast/api/service/JobBatchService.java

```diff
@@ -291,106 +151,23 @@ public Job createJobInNewTransaction(JobCreationDto jobDto) {
         jobService.updateJobStatus(jobId, status);
     }
 
-    /**
-     * Sets the final status of the batch once all jobs have been processed.
-     */
-    private void finalizeBatch(Long batchId, int total, int completed, int failed) {
-        BatchStatus finalStatus = (failed == total)
-                ? BatchStatus.FAILED
-                : (failed > 0) ? BatchStatus.COMPLETED_WITH_ERRORS : BatchStatus.COMPLETED;
-        updateBatchStatusInNewTransaction(batchId, finalStatus, total, completed, failed);
-        LOGGER.info(
-                "Batch {} processing finalized. Status: {}, Total: {}, Completed: {}, Failed: {}",
-                batchId,
-                finalStatus,
-                total,
-                completed,
-                failed);
-    }
-
-    /**
-     * Updates batch progress in a new transaction
-     */
-    @Transactional
-    public void updateBatchProgressInNewTransaction(@Nonnull Long batchId, int completedJobs, int failedJobs) {
-        try {
-            JobBatch batch = jobBatchRepository.findById(batchId);
-            if (batch != null) {
-                batch.setCompletedJobs(completedJobs);
-                batch.setFailedJobs(failedJobs);
-                jobBatchRepository.persist(batch);
-                LOGGER.info("Updated batch {} progress: completed={}, failed={}", batchId, completedJobs, failedJobs);
-            }
-        } catch (Exception e) {
-            LOGGER.error("Failed to update batch progress for batch {}", batchId, e);
-        }
-    }
-
-    /**
-     * Updates batch status in a new transaction to handle errors properly
-     */
-    @Transactional
-    public void updateBatchStatusInNewTransaction(
-            @Nonnull Long batchId, @Nonnull BatchStatus status, int totalJobs, int completedJobs, int failedJobs) {
-        try {
-            JobBatch batch = jobBatchRepository.findById(batchId);
-            if (batch != null) {
-                batch.setStatus(status);
-                batch.setTotalJobs(totalJobs);
-                batch.setCompletedJobs(completedJobs);
-                batch.setFailedJobs(failedJobs);
-                jobBatchRepository.persist(batch);
-                LOGGER.info("Updated batch {} final status: {}", batchId, status);
-            }
-        } catch (Exception e) {
-            LOGGER.error("Failed to update batch status for batch {}", batchId, e);
-        }
-    }
-
-    @Transactional
-    public void updateBatchTotalJobs(@Nonnull Long batchId, int totalJobs) {
-        JobBatch batch = jobBatchRepository.findById(batchId);
-        if (batch != null) {
-            batch.setTotalJobs(totalJobs);
-            jobBatchRepository.persist(batch);
-        }
-    }
-
     public List<JobBatchResponseDto> getAllBatches(int page, int size) {
-        return jobBatchRepository.findAll().page(Page.of(page, size)).list().stream()
+        return jobBatchRepository
+                .find("pipelineName", PipelineConstants.SAST_AI_PIPELINE)
```

Comment:
add something like below into our jobBatchRepository will easy the code further.

```
@ApplicationScoped
public class JobBatchRepository implements PanacheRepository<JobBatch> {

    public List<JobBatch> findByPipelineName(ExecutionMode name) {
        return list("pipelineName", name.getValue());
    }
}
```